spec/vector_spec.rb

```diff
@@ -449,6 +449,97 @@
               index: @multi_index, name: :precise_assignment, dtype: dtype))
           end
         end
+        
+        context Daru::CategoricalIndex do
+          context "non-numerical index" do
+            let (:idx) { Daru::CategoricalIndex.new [:a, :b, :a, :a, :c] }
+            let (:dv)  { Daru::Vector.new 'a'..'e', index: idx1 }
+
+            context "single category" do
+              context "multiple instances" do
+                subject { dv }
+                dv[:a] = 'x'
```

Comment:
This will not work (specs logic, I mean). You should place this assignment in `before{}` block:
```ruby
subject { dv }
before { dv[:a] = 'x' }
it  { ... }
```